Update release.yml

davidwaroquiers opened the pull request:
Testing to replace RELEASE_PAT_BOT by GITHUB_TOKEN

davidwaroquiers committed:
Update release.yml

Testing to replace RELEASE_PAT_BOT by GITHUB_TOKEN

ml-evs reviewed (commented):
I think this likely needs something like:

```
    permissions:
      contents: write
```

so that it can commit changes to the changelog. More at https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/controlling-permissions-for-github_token

davidwaroquiers committed:
Update release.yml

Added write permission for contents. Likely we need to iterate on this to check if maybe some other permissions are needed.
